src/components/gitProvider/GitProvider.tsx

```diff
@@ -1,41 +1,135 @@
-        <div className="form__subtitle">Manage your organization’s git accounts. &nbsp;
-            <a className="learn-more__href" href={DOCUMENTATION.GLOBAL_CONFIG_GIT} rel="noopener noreferrer" target="_blank">
-                Learn more about git accounts
-            </a>
-        </div>
-        {[{ id: null, name: "", active: true, url: "", authMode: "ANONYMOUS" }].concat(result && Array.isArray(result.result) ? result.result : []).sort((a, b) => a.name.localeCompare(b.name)).map(git => <CollapsedList {...git} key={git.id || Math.random().toString(36).substr(2, 5)} reload={reload} />)}
-    </section>
+    // if (loading && !result) return <Progressing pageLoader />
+    // if (error) {
+    //     showError(error)
+    //     if (!result) return null
+    // }
+
+    let allProviders = [{ id: null, name: "", active: true, url: "", gitHostId: "", authMode: "ANONYMOUS", userName: "", password: "" }].concat(providerList);
+
+    return (
+        <section className="mt-16 mb-16 ml-20 mr-20 global-configuration__component flex-1">
+            <h2 className="form__title">Git accounts</h2>
+            <div className="form__subtitle">Manage your organization’s git accounts. &nbsp;
+                <a className="learn-more__href" href={DOCUMENTATION.GLOBAL_CONFIG_GIT} rel="noopener noreferrer" target="_blank">
+                    Learn more about git accounts
+                </a>
+            </div>
+            {allProviders.map((provider) => {
+                return <CollapsedList key={provider.name || Math.random().toString(36).substr(2, 5)}
+                    id={provider.id}
+                    name={provider.name}
+                    hostList={hostList}
+                    providerList={providerList}
+                    hostListOption={hostListOption}
+                    gitHostId={provider.gitHostId}
+                    active={provider.active}
+                    url={provider.url}
+                    authMode={provider.authMode}
+                    userName={provider.userName}
+                    password={provider.password}
+                    getHostList={getHostList}
+                    getProviderList={getProviderList}
+                    reload={getInitData}
+                />
+            })}
+            {/* {[{ id: null, name: "", active: true, url: "", authMode: "ANONYMOUS" }].concat(result && Array.isArray(result.result) ? result.result : []).sort((a, b) => a.name.localeCompare(b.name)).map(git => <CollapsedList {...git} key={git.id || Math.random().toString(36).substr(2, 5)} reload={reload} />)} */}
+        </section>
     )
 }
 
-function CollapsedList({ id, name, active, url, authMode, accessToken = "", userName = "", password = "", reload, ...props }) {
+function CollapsedList({ id, name, active, url, authMode, gitHostId, accessToken = "", userName = "", password = "", reload, hostListOption, hostList, getHostList, getProviderList, providerList, ...props }) {
```

Comment:
too many input arguments should be avoided.. they should be wrapped up in an object in such cases